system/security/shared/secmanagertracedecorator.hpp

```diff
@@ -0,0 +1,364 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2024 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#pragma once
+
+#include "seclib.hpp"
+#include "jscm.hpp"
+#include "jthread.hpp"
+#include "jtrace.hpp"
+
+/**
+ * @brief Templated utility class to decorate an object with additional functionality.
+ *
+ * A decorator implements the same named interface as the objects it decorates. This class extends
+ * the interface, leaving implementation to subclasses. This class requires the decorated object to
+ * implement a named interface, even if only IInterface.
+ *
+ * In the ideal situation, the decorated object's interface is described completely by a named
+ * interface. By implementing the same interface, a decorator is interchangeable with the object
+ * it decoratos.
+ *
+ * In less than ideal situations, the decorated object's interface is an extension of a named
```

Comment:
Trivial, should "an extension" be "a subclass"

Reply:
I think they're synonyms, but I'll change it.